### PR TITLE
fix example

### PR DESCRIPTION
**Pull request by szha**
fixed multi-task learning example which wasn't running.
removed multiple copies of mnist_iterator and moved away from wget/unzip for portability.

### CODE REVIEW

**Comment by szha** [comment]
@zhreshold

File: example/multi-task/example_multi_task.py [hunk 1]
```diff
@@ -76,7 +79,13 @@ class Multi_Accuracy(mx.metric.EvalMetric):
     """Calculate accuracies of multi label"""
 
     def __init__(self, num=None):
-        super(Multi_Accuracy, self).__init__('multi-accuracy', num)
+        self.num = num
```
Comment (zhreshold) [inline example/multi-task/example_multi_task.py]: Better swap line 82 and 83.

Comment (szha) [reply example/multi-task/example_multi_task.py]: Parent constructor calls reset, which relies on self.num which doesn't exist if I put it as the first line.

File: example/numpy-ops/custom_softmax.py [hunk 2]
```diff
@@ -16,8 +16,12 @@
 # under the License.
 
 # pylint: skip-file
+import sys
 import os
-from data import mnist_iterator
+sys.path.insert(0, "../../python/")
```
[the remark this answers is not in the document]
Comment (szha) [reply example/numpy-ops/custom_softmax.py]: There is a get_mnist() in test_utils, though the construction is different. In this PR I just intend to fix the broken example for multi-task and remove several duplicate copies of mnist along the way.

**Pull request merged**